### PR TITLE
Bundle the clj-kondo extension

### LINKED ISSUE #423: Bundle the clj-kondo extension
Calva's linting support will never be close to as good as the [clj-kondo extension](https://github.com/borkdude/clj-kondo.lsp), which just works without any installation of binaries by the user, and generally just is wonderfully helpful.

If you can't beat them, join them, is the saying, right? Calva should bundle the clj.kondo extension! (And it will, soon).

### PR DESCRIPTION
**Pull request by PEZ**
## What has Changed?
- All listing code is evicted from Calva
- `borkdude.clj-kondo` added to the Calva `extensionPack`.
- Documentation updated.

Fixes #423 

## My Calva PR Checklist

I have:

- [x] Read [How to Contribute](https://github.com/BetterThanTomorrow/calva/wiki/How-to-Contribute#before-sending-pull-requests).
- [x] Made sure I am directing this pull request at the `dev` branch. (Or have specific reasons to target some other branch.)
- [x] Made sure I am changed the default PR base branch, so that it is not `master`. (Sorry for the nagging.)
- [x] Tested the VSIX built from the PR (well, if this is a PR that changes the source code.) You'll find the artifacts by clicking _Show all checks_ in the CI section of the PR page, and then _Details_ on the `ci/circleci: build` test. (For now you'll need to opt in to the CircleCI _New Experience_ UI to see the Artifacts tab, because bug.)
     - [x] Tested the particular change
     - [x] Figured if the change might have some side effects and tested those as well.
     - [x] Smoke tested the extension as such.
- [x] Referenced the issue I am fixing/addressing _in a commit message for the pull request_.
     - [x] If I am fixing the issue, I have used [GitHub's fixes/closes syntax](https://help.github.com/en/articles/closing-issues-using-keywords)
- [x] Created the issue I am fixing/addressing, if it was not present.
- [x] Updated the `[Unreleased]` entry in `CHANGELOG.md`, linking the issue(s) that the PR is addressing.

## The Calva Team PR Checklist:

Before merging we (at least one of us) have:

- [ ] Made sure the PR is directed at the `dev` branch (unless reasons).
- [ ] Read the source changes.
- [ ] Given feedback and guidance on source changes, if needed. (Please consider noting extra nice stuff as well.)
- [ ] Tested the VSIX built from the PR (well, if this is a PR that changes the source code.)
     - [ ] Tested the particular change
     - [ ] Figured if the change might have some side effects and tested those as well.
     - [ ] Smoke tested the extension as such.
- [ ] If need be, had a chat within the team about particular changes.

Ping  @kstehn, @cfehse, @borkdude, @agoldhammer

### CODE REVIEW

**Comment by borkdude** [comment]
I would rephrase this a little bit:

> One such error could be from not knowing this: Note that the symbols in the clj-kondo configuration need to be fully qualified, with namespaces.
that's not always true, depends on the config per linter, but it's clear from the README

It depends on the linter if symbols are fully qualified. Some linters expect maps with namespaces to single symbols for example. See https://github.com/borkdude/clj-kondo/blob/master/doc/config.md.

**Comment by PEZ** [comment]
There. Fixed it. 😄
